src/test/java/com/squareup/javapoet/TypesTest.java

```diff
@@ -78,36 +77,34 @@ private TypeMirror getType(Class<?> clazz) {
     assertThat(Types.get(typeVariables.get(0).asType()))
         .isEqualTo(Types.typeVariable("Simple"));
     assertThat(Types.get(typeVariables.get(1).asType()))
-        .isEqualTo(Types.typeVariable("ExtendsClass", Number.class));
+        .isEqualTo(Types.typeVariable("ExtendsClass", ClassName.get(Number.class)));
     assertThat(Types.get(typeVariables.get(2).asType()))
-        .isEqualTo(Types.typeVariable("ExtendsInterface", Runnable.class));
+        .isEqualTo(Types.typeVariable("ExtendsInterface", ClassName.get(Runnable.class)));
     assertThat(Types.get(typeVariables.get(3).asType()))
         .isEqualTo(Types.typeVariable("ExtendsTypeVariable", Types.typeVariable("Simple")));
-    assertThat(Types.get(typeVariables.get(4).asType()))
-        .isEqualTo(Types.typeVariable("Intersection", Number.class, Runnable.class));
-    assertThat(Types.get(typeVariables.get(5).asType()))
-        .isEqualTo(Types.typeVariable("IntersectionOfInterfaces",
-            Runnable.class, Serializable.class));
+    assertThat(Types.get(typeVariables.get(4).asType())).isEqualTo(
+        Types.typeVariable("Intersection",
+            ClassName.get(Number.class), ClassName.get(Runnable.class)));
```

Comment:
Yup, it's because equals changed. Added a comment. The `toString()` is already covered.